editor/src/clj/editor/code/shader.clj

```diff
@@ -168,35 +175,88 @@
         {:resource resource
          :content content}))))
 
-(g/defnk produce-build-targets [_node-id compile-spirv lines resource]
+(g/defnk produce-build-targets [_node-id compile-spirv proj-path+full-lines resource]
   [(bt/with-content-hash
      {:node-id _node-id
       :resource (workspace/make-build-resource resource)
       :build-fn build-shader
       :user-data {:compile-spirv compile-spirv
-                  :lines lines
+                  :lines (second proj-path+full-lines)
                   :resource-ext (resource/type-ext resource)}})])
 
+(def ^:private include-pattern #"^\s*#include \"([^\"]+?)\"\s*$")
```

Comment:
Are extra spaces between `#include` and `"` allowed? Are comments after the second `"` allowed? E.g.:
```glsl
#include  "foo.glsl"  // needed for foo function
```